File: web/src/types/product.ts
export type Product = {
  id?: string;
  sku: string;
  name: string;
  description: string;
  category: string;
  price: number;
  stock: number;
  active: boolean;
};

export type ProductsPage = {
  content: Product[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
};

File: api/src/main/java/com/products/domain/model/Product.java
package com.products.domain.model;

import com.products.domain.model.BaseEntity;
import io.quarkus.mongodb.panache.common.MongoEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@MongoEntity(collection = "products")
public class Product extends BaseEntity {

    public String sku;
    public String name;
    public String description;
    public String category;
    public Double price;
    public Integer stock;
    public Boolean active = true;

    public void create(String user) {
        this.markCreated(user);
    }

    public boolean hasStock() {
        return stock != null && stock > 0;
    }

    public boolean isAvailable() {
        return Boolean.TRUE.equals(active) && hasStock();
    }
}